examples/graphics_context.zig

```diff
@@ -143,8 +143,7 @@ pub const GraphicsContext = struct {
     }
 
     pub fn deviceName(self: *const GraphicsContext) []const u8 {
-        const len = std.mem.indexOfScalar(u8, &self.props.device_name, 0).?;
-        return self.props.device_name[0..len];
+        return &self.props.device_name;
```

Comment:
Sorry for not noticing this earlier, but this function is used in triangle.zig to print the device name. Without sliceTo here it will attempt to print a bunch of undefined data, so can you please add sliceTo here too? Note that `[:0]const u8` is not valid here because if the device name uses all 64 bytes then there is no zero terminator.

Reply:
why it does work fine without sliceTo?
I mean it does print the device name correctly, am I missing something?

Reply:
Its possible that the remaining bytes are initialized to something that is not printable, for examle null bytes. What happens if you print the name in hex?

Reply:
it printed this:
Using device: { 78, 86, 73, 68, 73, 65, 32, 71, 101, 70, 111, 114, 99, 101, 32, 71, 84, 88, 32, 49, 54, 53, 48, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }

Reply:
Right, so I dont think that this function should return the null bytes. Afaik there is no requirements to initialize the bytes after the first null byte.

Reply:
done